src/groupeddataframe/fastaggregates.jl

```diff
@@ -17,20 +17,12 @@ check_aggregate(f::typeof(prod), ::AbstractVector{<:Union{Missing, Number}}) =
     Reduce(Base.mul_prod)
 check_aggregate(f::typeof(prod∘skipmissing), ::AbstractVector{<:Union{Missing, Number}}) =
     Reduce(Base.mul_prod, !ismissing)
```

Comment:
@nalimilan - I do not understand why you added these methods (if I remember correctly you added them when creating `check_aggregate`, but maybe it was me 😄). Apart from creating dispatch ambiguities (which lead to a bug) - they seem not to be needed, as they are captured by the default definition of `check_aggregate`.

Reply:
Well I put the blame on you as these come from https://github.com/JuliaData/DataFrames.jl/pull/2357/! But I guess I'm guilty of not seeing this when reviewing. :-)

Reply:
But I understand you are OK with the fix?

Reply:
I agree these methods seem redundant with the fallbacks, and we have tests for  spotted at https://github.com/JuliaData/DataFrames.jl/pull/2357, right?

Reply:
The tests would not catch anything. I do not know why I added these methods. The point is that:
```
check_aggregate(f::typeof(maximum),
                ::AbstractVector{<:Union{Missing, MULTI_COLS_TYPE, AbstractVector}}) = f
```
is a signature when someone wants to compute `maximum` on a column that is e.g. a vector of data frames, or a vector of vectors. Apart from the fact that it is super rate, it should just use the default definition:
```
check_aggregate(f::Any, ::AbstractVector) = f
```
which has the same behavior.

In summary - in my opinion it is safe to just remove it.

The specialized method is:
```
check_aggregate(f::typeof(maximum), v::AbstractVector{<:Union{Missing, Real}}) =
    eltype(v) === Any ? f : Reduce(max)
```
which for sure will not cause problem. This lead me to noticing that the `eltype(v)` check is also not needed there as clearly `AbstractVector{<:Union{Missing, Real}}` cannot have `Any` `eltype`. I have pushed a fix to this.